test/checkpointing/test_checkpointing.py

```diff
@@ -0,0 +1,29 @@
+import pytest
+import os
+from src.checkpointing import Checkpoint
+
+@pytest.fixture
+def checkpoint():
+    checkpoint = Checkpoint()
+    checkpoint._file = "checkpoint.txt"  # Kinda gross but should make it easy to determine the output
+    return checkpoint
```

Comment:
You should use pytest's fixture `tmp_path` to do this instead of using an actual file.
Something like
```
def checkpoint(tmp_path):
    checkpoint = Checkpoint()
    checkpoint._file = tmp_path / "checkpoint.txt"
    return checkpoint
```
That should eliminate having to write a temp file onto the test directory.